src/supertux/shrinkfade.cpp

```diff
@@ -41,9 +42,9 @@ void
 ShrinkFade::draw(DrawingContext& context)
 {
   float progress = m_accum_time / m_fade_time;
-  float diameter = 2 * m_initial_size * (1.0f - progress);
+  float diameter = 2 * m_initial_size * (m_direction == FADEOUT ? (1.0f - progress) : progress);
   context.color().draw_inverse_ellipse(m_dest, Vector(1.1f * diameter, diameter),
-                                         Color(0, 0, 0), LAYER_GUI + 1);
+                                         Color(0, 0, 0), LAYER_LIGHTMAP - 1);
```

Comment:
These transitions are not only used in levels, but also in the GUI as well. I wonder if we could use a variable here to know which is the case?